### DISCUSSION TITLE
writeUnsignedByte method for ByteArray

### DISCUSSION THREAD

**Discussion by manishpatel0404** [airsdk/Adobe-Runtime-Support#2338] [Q&A] [answered]
Is a `writeUnsignedByte` method available on `ByteArray` class? If not, is there a way to write an unsigned byte to a ByteArray?

Thanks.

**Comment by ajwfrost** [comment, marked as answer]
You can just use `writeByte`, there's no need for the runtime to distinguish between signed and unsigned at this level because what you're actually passing in is an `int` type.. 

The documentation states:
> The low 8 bits of the parameter are used. The high 24 bits are ignored.
> value:int — A 32-bit integer. The low 8 bits are written to the byte stream.

So if your int value was 0x180 you'd write 0x80 to the stream, if it was 0x17F you'd write 0x7F to the stream, etc.  A signed '-1' = 0xFFFFFFFF would end up with 0xFF in the stream, but an unsigned '4294967295' would also end up with 0xFF in the stream.

The bit where you actually have to worry about the sign is when you read back, from the byte into an application:

`readByte` returns an integer between -128 and 127, i.e. if 0x80 is read back, it's sign-extended into the `int` value of 0xFFFFFF80 = -128.
`readUnsignedByte` returns an unsigned integer between 0 and 256, i.e. if 0x80 is read back, it's not sign-extended, it becomes 0x00000080 = 128.

Hope that helps..

**Comment by manishpatel0404** [comment]
Thanks @ajwfrost 

Changed my code to `readUnsignedByte` instead of `readByte`. I had misunderstood the documentation. I now realise that `writeUnsignedByte` would be redundant in this case.
